File: sdks/python/bolt_prediction_sdk/models/health_index_detail_schema.py
"""
HealthIndexDetailSchema 模型
"""

from typing import Optional, Any, Dict, List
from datetime import datetime
from pydantic import Field

from .base import SDKBaseModel


class HealthIndexDetailSchema(SDKBaseModel):
    """健康度指数详情"""

    hi_score: float = Field(description="综合健康度指数 0-100")
    hi_level: str = Field(description="健康等级 excellent/good/fair/poor/critical")
    factors: List[HealthIndexFactorSchema] = Field(description="各因子得分详情")
    preload_stability_score: float = Field(description="预紧力稳定性得分")
    alert_frequency_score: float = Field(description="预警频率得分")
    fault_history_score: float = Field(description="故障历史得分")
    environmental_stress_score: float = Field(description="环境应力得分")
    service_age_score: float = Field(description="使用年限得分")
    trend: Optional[Any] = Field(description="健康趋势 improving/stable/declining", default=None)
    trend_rate: Optional[Any] = Field(description="趋势变化率", default=None)
    calculate_time: datetime = Field()

File: sdks/java/src/main/java/com/boltprediction/sdk/model/HealthIndexDetailSchema.java
package com.boltprediction.sdk.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import java.util.List;
import java.util.Map;
import java.time.OffsetDateTime;

/** 健康度指数详情 */
@JsonIgnoreProperties(ignoreUnknown = true)
public class HealthIndexDetailSchema {

    @JsonProperty("hi_score")
    private Double hiScore;

    @JsonProperty("hi_level")
    private String hiLevel;

    @JsonProperty("factors")
    private List<HealthIndexFactorSchema> factors;

    @JsonProperty("preload_stability_score")
    private Double preloadStabilityScore;

    @JsonProperty("alert_frequency_score")
    private Double alertFrequencyScore;

    @JsonProperty("fault_history_score")
    private Double faultHistoryScore;

    @JsonProperty("environmental_stress_score")
    private Double environmentalStressScore;

    @JsonProperty("service_age_score")
    private Double serviceAgeScore;

    @JsonProperty("trend")
    private Object trend;

    @JsonProperty("trend_rate")
    private Object trendRate;

    @JsonProperty("calculate_time")
    private OffsetDateTime calculateTime;

    public Double getHiScore() {
        return hiScore;
    }

    public void setHiScore(Double hiScore) {
        this.hiScore = hiScore;
    }

    public String getHiLevel() {
        return hiLevel;
    }

    public void setHiLevel(String hiLevel) {
        this.hiLevel = hiLevel;
    }

    public List<HealthIndexFactorSchema> getFactors() {
        return factors;
    }

    public void setFactors(List<HealthIndexFactorSchema> factors) {
        this.factors = factors;
    }

    public Double getPreloadStabilityScore() {
        return preloadStabilityScore;
    }

    public void setPreloadStabilityScore(Double preloadStabilityScore) {
        this.preloadStabilityScore = preloadStabilityScore;
    }

    public Double getAlertFrequencyScore() {
        return alertFrequencyScore;
    }

    public void setAlertFrequencyScore(Double alertFrequencyScore) {
        this.alertFrequencyScore = alertFrequencyScore;
    }

    public Double getFaultHistoryScore() {
        return faultHistoryScore;
    }

    public void setFaultHistoryScore(Double faultHistoryScore) {
        this.faultHistoryScore = faultHistoryScore;
    }

    public Double getEnvironmentalStressScore() {
        return environmentalStressScore;
    }

    public void setEnvironmentalStressScore(Double environmentalStressScore) {
        this.environmentalStressScore = environmentalStressScore;
    }

    public Double getServiceAgeScore() {
        return serviceAgeScore;
    }

    public void setServiceAgeScore(Double serviceAgeScore) {
        this.serviceAgeScore = serviceAgeScore;
    }

    public Object getTrend() {
        return trend;
    }

    public void setTrend(Object trend) {
        this.trend = trend;
    }

    public Object getTrendRate() {
        return trendRate;
    }

    public void setTrendRate(Object trendRate) {
        this.trendRate = trendRate;
    }

    public OffsetDateTime getCalculateTime() {
        return calculateTime;
    }

    public void setCalculateTime(OffsetDateTime calculateTime) {
        this.calculateTime = calculateTime;
    }

}